### PR TITLE
feat(runWindows): Adds option to start application with remote debugging

### PR DESCRIPTION
**Pull request by rozele**
Tools like VS Code need to be able to launch the app from the command line with remote debugging enabled.  This change allows arguments to be passed from powershell to the app and down to the DevSupportManager.

### CODE REVIEW

File: ReactWindows/ReactNative/DevSupport/WebSocketJavaScriptExecutor.cs [hunk 1]
```diff
@@ -190,9 +195,40 @@ public void Dispose()
...
-            // TODO: check result of `StoreAsync()`
+            if (!_isDisposed)
+            {
+                _messageWriter.WriteString(message);
+                await _messageWriter.StoreAsync();
+                // TODO: check result of `StoreAsync()`
+            }
+            else
+            {
+                DoCallback(requestId, null);
```
Comment (rozele) [inline ReactWindows/ReactNative/DevSupport/WebSocketJavaScriptExecutor.cs]: Cancel?

File: local-cli/runWindows/utils/WindowsStoreAppUtils.ps1 [hunk 2]
```diff
@@ -150,9 +150,12 @@ function Install-AppFromAppx {
 function Start-Locally {
     param(
         [Parameter(Mandatory=$true, Position=0, ValueFromPipelineByPropertyName=$true)]
-        [string] $ID <# package.appxmanifest//Identity@name #>
-    )
+        [string] $ID, <# package.appxmanifest//Identity@name #>
 
+        [Parameter(Mandatory=$true, Position=1, ValueFromPipelineByPropertyName=$true)]
+        [string] $proxy
```
Comment (rozele) [inline local-cli/runWindows/utils/WindowsStoreAppUtils.ps1]: Let's just thread the args through so this is a more general purpose utility (and make optional).

Comment (mpodwysocki) [inline local-cli/runWindows/utils/WindowsStoreAppUtils.ps1]: @rozele so instead of `$proxy`, you just have it as a string which is a list of args?

Comment (rozele) [inline local-cli/runWindows/utils/WindowsStoreAppUtils.ps1]: Yep.

---
In reply to: [68452196](https://github.com/ReactWindows/react-native-windows/pull/499#discussion_r68452196) [](ancestors = 68452196)

File: ReactWindows/ReactNative/DevSupport/WebSocketJavaScriptExecutor.cs (hunk 1 above)
Comment (rozele) [inline ReactWindows/ReactNative/DevSupport/WebSocketJavaScriptExecutor.cs]: Not going to fix this, trying to catch the exception and prevent it from going all the way to the redbox is too difficult for test code.

---
In reply to: [68448798](https://github.com/ReactWindows/react-native-windows/pull/499#discussion_r68448798) [](ancestors = 68448798)

File: ReactWindows/ReactNative/DevSupport/WebSocketJavaScriptExecutor.cs [hunk 3]
```diff
@@ -25,12 +26,15 @@ class WebSocketJavaScriptExecutor : IJavaScriptExecutor
         private bool _connected;
         private DataWriter _messageWriter;
         private int _requestId;
+
```
Comment (mpodwysocki) [inline ReactWindows/ReactNative/DevSupport/WebSocketJavaScriptExecutor.cs]: Why a line between other vars?

**Comment by mpodwysocki** [comment]
:shipit:

**Pull request merged**